### PR TITLE
Add option to gather code coverage for specific targets

### PR DESCRIPTION
**Pull request by kateinoigakukun**
To avoid conflict, this branch is based on https://github.com/yonaskolb/XcodeGen/pull/655 and https://github.com/tuist/XcodeProj/pull/473.

**Please merge https://github.com/yonaskolb/XcodeGen/pull/655 first :pray:**

I added `enableCoverage` option for each test target to gather code coverage for specific targets.

### CODE REVIEW

File: Sources/ProjectSpec/Scheme.swift [hunk 1]
```diff
@@ -115,6 +116,24 @@ public struct Scheme: Equatable {
         public var language: String?
         public var region: String?
 
+        public struct CoverageTarget: Equatable, ExpressibleByStringLiteral {
```
Comment (yonaskolb) [inline Sources/ProjectSpec/Scheme.swift]: This could be the `TargetReference` I mention in #655

**Comment by brandtdaniels** [comment]
Can we get some traction on this PR?

**Comment by yonaskolb** [comment]
This PR currently relies on https://github.com/yonaskolb/XcodeGen/pull/655. It would get merged a lot faster if it just included the new code coverage stuff

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 #### Added
 - Support Target Reference to another project. [#655](https://github.com/yonaskolb/XcodeGen/pull/655) @kateinoigakukun
-
+- Added `enableCoverage` for each test target. This enables to gather code coverage for specific targets. [#656](https://github.com/yonaskolb/XcodeGen/pull/656) @kateinoigakukun
```
Comment (yonaskolb) [inline CHANGELOG.md]: This might be outdated. The new property is coverageTargets

**Pull request merged**